src/DI/Config/Adapters/NeonAdapter.php

```diff
@@ -37,20 +37,21 @@ private function process(array $arr)
 	{
 		$res = [];
 		foreach ($arr as $key => $val) {
-			if (substr($key, -1) === self::PREVENT_MERGING) {
-				if (!is_array($val) && $val !== NULL) {
-					throw new Nette\InvalidStateException("Replacing operator is available only for arrays, item '$key' is not array.");
-				}
-				$key = substr($key, 0, -1);
-				$val[Helpers::EXTENDS_KEY] = Helpers::OVERWRITE;
-
-			} elseif (preg_match('#^(\S+)\s+' . self::INHERITING_SEPARATOR . '\s+(\S+)\z#', $key, $matches)) {
-				if (!is_array($val) && $val !== NULL) {
-					throw new Nette\InvalidStateException("Inheritance operator is available only for arrays, item '$key' is not array.");
-				}
-				list(, $key, $val[Helpers::EXTENDS_KEY]) = $matches;
-				if (isset($res[$key])) {
-					throw new Nette\InvalidStateException("Duplicated key '$key'.");
+			if (is_string($key)) {
```

Comment:
Please can you change the deep of nesting, ie. use `is_string($key)` in both conditions or `if (!is_string($key)))` as first condition?